prog/base.rb

```diff
@@ -166,10 +166,12 @@ def donate
 
   def reap
     exited_children = strand.children_dataset.where(Sequel.~(exitval: nil))
-    exited_children_count = exited_children.count
+    exited_children_materialized = DB.fetch(exited_children).all
+    exited_children_count = exited_children_materialized.count
     exited_children.destroy
     # Clear cache if anything was deleted.
     strand.associations.delete(:children) if exited_children_count > 0
+    exited_children_materialized
```

Comment:
I tried to replicate returning.delete behavior, which uses DB.fetch internally. As an alternative, we can simplify this code quite a bit by modifying the caller in a way to expect instantiated models instead of database rows. Let me quickly open a PR, so that we can compare.

Reply:
I think this turned out to be quite nice;
https://github.com/ubicloud/ubicloud/pull/404